lib/ruby_indexer/lib/ruby_indexer/prefix_tree.rb

```diff
@@ -0,0 +1,103 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RubyIndexer
+  class PrefixTree
+    extend T::Sig
+
+    sig { params(items: T::Array[String]).void }
+    def initialize(items)
+      @root = T.let(Node.new("", ""), Node)
+
+      items.each { |item| insert(item) }
+    end
+
+    sig { params(prefix: String).returns(T::Array[String]) }
+    def search(prefix)
+      node = T.let(@root, Node)
```

Comment:
I'm surprised we need this `T.let`, do we ever change the type of `node` in the loop?